apps/studio/data/ai/cron-syntax-mutation.ts

```diff
@@ -0,0 +1,52 @@
+import { useMutation, UseMutationOptions } from '@tanstack/react-query'
+import { toast } from 'sonner'
+
+import { isResponseOk, post } from 'lib/common/fetch'
+import { BASE_PATH } from 'lib/constants'
+import type { ResponseError } from 'types'
+
+export type CronSyntaxGenerateVariables = {
+  prompt: string
+}
+
+export async function generateCronSyntax({ prompt }: CronSyntaxGenerateVariables) {
+  const response = await post<string>(BASE_PATH + '/api/ai/sql/cron', {
```

Comment:
just remember we need to add this to `middleware.ts`

and i reckon we should also try to figure out how to just cover all API routes under /api/ai/sql in there too - guessing its likely all endpoints under there will be used by hosted too

Reply:
sorry, i've got this in here — is there something missing?

https://github.com/supabase/supabase/pull/30345/files#diff-3aeded5f052bc3f0f64fa6ba9db7afacecc1b07ad9df802c82798a31679baede

Reply:
ahh yes that will do!